lib/ansible/modules/network/routeros/routeros_facts.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/python
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: routeros_facts
+version_added: "2.8"
+author: "Egor Zaitsev (@heuels)"
+short_description: Collect facts from remote devices running MikroTik RouterOS
+description:
+  - Collects a base set of device facts from a remote device that
+    is running RotuerOS.  This module prepends all of the
+    base network fact keys with C(ansible_net_<fact>).  The facts
+    module will always collect a base set of facts from the device
+    and can enable or disable collection of additional facts.
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        C(all), C(hardware), C(config), and C(interfaces).  Can specify a list of
+        values to include a larger subset.  Values can also be used
+        with an initial C(!) to specify that a specific subset should
+        not be collected.
+    required: false
+    default: '!config'
+"""
+
+EXAMPLES = """
+# Collect all facts from the device
+- routeros_facts:
+    gather_subset: all
+
+# Collect only the config and default facts
+- routeros_facts:
+    gather_subset:
+      - config
+
+# Do not collect hardware facts
+- routeros_facts:
+    gather_subset:
+      - "!hardware"
+"""
+
+RETURN = """
+ansible_net_gather_subset:
+  description: The list of fact subsets collected from the device
+  returned: always
+  type: list
+
+# default
+ansible_net_model:
+  description: The model name returned from the device
+  returned: always
+  type: string
+ansible_net_serialnum:
+  description: The serial number of the remote device
+  returned: always
+  type: string
+ansible_net_version:
+  description: The operating system version running on the remote device
+  returned: always
+  type: string
+ansible_net_hostname:
+  description: The configured hostname of the device
+  returned: always
+  type: string
+
+# hardware
+ansible_net_spacefree_mb:
+  description: The available disk space on the remote device in Mb
```

Comment:
It's probably better to use MiB instead of Mb or MB. (Here and few more times below.)